src/librbd/ImageWatcher.cc

```diff
@@ -844,7 +844,6 @@ void ImageWatcher::handle_payload(const FlattenPayload &payload,
 
 void ImageWatcher::handle_payload(const ResizePayload &payload,
 				  bufferlist *out) {
-  RWLock::RLocker l(m_image_ctx.owner_lock);
```

Comment:
Here is my thought. Correct me if I'm wrong.

When ownership is lost, all the AsyncRequests are marked as cancelled.But it is possible that in the callback of a middle step of the AsyncRequest state machine, it is marked as cancelled after the cancellation check and before it takes the owner lock in the next step.
Take AsyncResizeRequest for example. let's assume it's in step STATE_UPDATE_HEADER and doing async update header. Later when it calls back, the `AsyncRequest::complete` function checks if it's cancelled. If not, it calls `should_complete` and goes to the next step 'shrink object map'. If the ownership is lost between the cancellation check and when taking the owner lock in `send_shrink_object_map`, the lost ownership check in this function won't be an assert.